theano/tensor/blas.py

```diff
@@ -319,7 +322,9 @@ def default_blas_ldflags():
     # readily available. We try to see if that's the case, rather
     # than disable blas. To test it correctly, we must load a program.
     # Otherwise, there could be problem in the LD_LIBRARY_PATH.
-    return try_blas_flag(['-lblas'])
+    ret = try_blas_flag(['-lblas'])
+    if ret:
+        return ' '.join(ret)
```

Comment:
Which is what this code does.  But it doesn't explain the failure because the previous code never worked.